subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -574,7 +573,9 @@ static void encode_action_status(struct bt_mesh_scheduler_srv *srv,
 	if (is_reduced) {
 		net_buf_simple_add_u8(buf, idx);
 	} else {
-		scheduler_action_pack(buf, idx, &srv->sch_reg[idx]);
+		if (!scheduler_action_pack(buf, idx, &srv->sch_reg[idx])) {
+			LOG_WRN("Scheduler action status has invalid entry values");
```

Comment:
It should never happen here.

Reply:
I agree. But is it better to make use of the return value just in case?

Reply:
I do not think checking is required here at all. It has been already checked in `action_set`. Why do we need to check it twice?

Reply:
I would leave packing\unpacking as is. Just add sanitizing function and be flexible to use it or to not use it in right places.
The splitting functionality on minimum valuable functions gives you flexibility in usage.
Really do not see any reason to attach sanitizing into parser\composer.